### PR TITLE
add ability to choose indefinite date on calendar

### PR DESCRIPTION
**Pull request by Sammuel09**
image         image

[Link to figma](https://www.figma.com/file/WwEmXK4bXgFQIDwCfWOXRT/Recurring-Billing?type=design&node-id=949-92235&mode=design&t=qw7OkLH65WykYC5s-4)

## Summary by CodeRabbit

- **New Features**
	- Introduced an `isIndefinite` option in Calendar and DatePicker components, allowing users to select an indefinite date range.
	- Added a visual representation of the selected model value in Checkbox components for enhanced user feedback.

- **Enhancements**
	- Improved calendar functionality with new logic to support 'Indefinite' selections, making date picking more versatile.
	- Updated DatePicker component to handle 'Indefinite' values more effectively and to display dates in "DD/MM/YYYY" format.

### CODE REVIEW

File: packages/vue/src/Form/Calendar/OcCalendar.vue [hunk 1]
```diff
@@ -225,6 +234,13 @@ const doneSelecting = () => {
       : selectedStartDate.value.toDate(),
   );
 };
+
+const isCalendarIndefinite = ref(false);
+const handleIndefinite = (value) => {
+  console.log('value', value)
```
Comment (kutsaniuk) [inline packages/vue/src/Form/Calendar/OcCalendar.vue:240]: Please remove `console.log`

File: packages/vue/src/Form/Calendar/OcCalendar.vue [hunk 2]
```diff
@@ -225,6 +234,13 @@ const doneSelecting = () => {
...
+  emit("update:isIndefinite", value);
+  // isCalendarIndefinite.value = value;
```
Comment (kutsaniuk) [inline packages/vue/src/Form/Calendar/OcCalendar.vue:242]: Please remove comment

File: packages/vue/src/Form/DatePicker/DatePicker.vue [hunk 3]
```diff
@@ -111,6 +115,25 @@ const resetCalendar = () => {
...
+  const isInCurrentMonth = (date) => date.get('month') === dayjs().get('month');
+  return isInCurrentMonth(date);
+}
+
+const isCalendarIndefinite = ref(false);
+
+const handleIndefinite = (event) => {
+  console.log('event from calendar', event)
+  if (event) {
+    // emit("update:modelValue", 'Indefinite');
```
Comment (kutsaniuk) [inline packages/vue/src/Form/DatePicker/DatePicker.vue:130]: Please remove comment and `console.log`

File: packages/vue/src/Form/DatePicker/DatePicker.vue [hunk 4]
```diff
@@ -111,6 +115,25 @@ const resetCalendar = () => {
...
+    isCalendarIndefinite.value = true;
+  } else {
+    isCalendarIndefinite.value = false;
+  }
+  event ? emit("update:modelValue", 'Indefinite') : emit("update:modelValue", null);
```
Comment (kutsaniuk) [inline packages/vue/src/Form/DatePicker/DatePicker.vue:135]:
```suggestion
  emit("update:modelValue", event ?  "Indefinite" : null);
```

**Review by kutsaniuk — APPROVED** [review]
LGTM!

File: packages/vue/src/Form/DatePicker/DatePicker.vue [hunk 5]
```diff
@@ -111,6 +115,25 @@ const resetCalendar = () => {
...
+  return isInCurrentMonth(date);
+}
+
+const isCalendarIndefinite = ref(false);
+
+const handleIndefinite = (event) => {
+  console.log('event from calendar', event)
+  if (event) {
+    // emit("update:modelValue", 'Indefinite');
+    isCalendarIndefinite.value = true;
```
Comment (kutsaniuk) [inline packages/vue/src/Form/DatePicker/DatePicker.vue:131]:
```suggestion
    isCalendarIndefinite.value = !!event
```

File: packages/vue/src/Form/DatePicker/DatePicker.vue [hunk 6]
```diff
@@ -111,6 +115,25 @@ const resetCalendar = () => {
...
+  } else {
+    isCalendarIndefinite.value = false;
```
Comment (kutsaniuk) [inline packages/vue/src/Form/DatePicker/DatePicker.vue:133]:
```suggestion
    isCalendarIndefinite.value = false;
```

Comment (kutsaniuk) [reply packages/vue/src/Form/DatePicker/DatePicker.vue:133]: And remove `if (event) {
    // emit("update:modelValue", 'Indefinite');
    isCalendarIndefinite.value = true;
  } else {
    isCalendarIndefinite.value = false;
  }`

File: packages/vue/src/Form/DatePicker/DatePicker.vue (hunk 4 above)
Comment (Sammuel09) [reply packages/vue/src/Form/DatePicker/DatePicker.vue:135]: ha. nice one! thanks !!

**Pull request merged**